shim/opentelemetry-opencensus-shim/src/opentelemetry/shim/opencensus/_patch.py

```diff
@@ -33,10 +33,15 @@ def install_shim(
         __version__,
         tracer_provider=tracer_provider,
     )
-    shim_tracer = ShimTracer(NoopTracer(), otel_tracer=otel_tracer)
 
-    def fget_tracer(self) -> ShimTracer:
-        return shim_tracer
+    def fget_tracer(self: Tracer) -> ShimTracer:
+        # self.span_context is how instrumentations pass propagated context into OpenCensus e.g.
+        # https://github.com/census-instrumentation/opencensus-python/blob/fd064f438c5e490d25b004ee2545be55d2e28679/contrib/opencensus-ext-flask/opencensus/ext/flask/flask_middleware.py#L147-L153
+        return ShimTracer(
```

Comment:
Do we need to create a new `ShimTracer` every time this property is accessed?

Reply:
Hmm we could cache it to be once per "`self`" Tracer instance. Otherwise we do need to pass in the wrapped span context on L42 so it can't be a global singleton anymore

Reply:
I added an lru_cache with the default maxsize which should help here without too much memory cost